presentation/src/main/java/com/knocklock/presentation/home/HomeViewModel.kt

```diff
@@ -12,34 +14,66 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
-    getLockScreenUseCase: GetLockScreenUseCase
+    getLockScreenUseCase: GetLockScreenUseCase,
+    private val saveWallPaperUseCase: SaveWallPaperUseCase
 ) : ViewModel() {
-    private val initHomeMenuList = listOf(HomeMenu.SETTING)
+    private val initHomeMenuList = listOf(HomeMenu.SETTING, HomeMenu.TMP, HomeMenu.SAVE)
 
     private val homeMenuList = MutableStateFlow(initHomeMenuList)
 
-    val homeScreenUiState = getLockScreenUseCase()
-        .combine(homeMenuList) { lockScreen, menuList ->
-            HomeScreenUiState.Success(
-                lockScreen = lockScreen,
-                menuList = menuList.toImmutableList()
-            )
-        }.stateIn(
-            scope = viewModelScope,
-            started = SharingStarted.WhileSubscribed(5_000),
-            initialValue = HomeScreenUiState.Loading,
+    private val tmpHomeScreen: MutableStateFlow<TmpScreenState> =
+        MutableStateFlow(TmpScreenState.None)
+
+    private val savedHomeScreenState = getLockScreenUseCase()
+
+    val homeScreenUiState = combine(
+        homeMenuList,
+        tmpHomeScreen,
+        savedHomeScreenState
+    ) { menuList, tmpScreen, savedScreen ->
+        HomeScreenUiState.Success(
+            menuList = menuList.toImmutableList(),
+            lockScreen = if (tmpScreen is TmpScreenState.Custom) tmpScreen.screen else savedScreen
         )
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5_000),
+        initialValue = HomeScreenUiState.Loading
+    )
+
+    fun saveWallPaper() {
+        viewModelScope.launch {
+            val value = tmpHomeScreen.value
+            if (value is TmpScreenState.Custom) {
+                saveWallPaperUseCase((value.screen.background as LockScreenBackground.LocalImage).imageUri)
+            }
```

Comment:
추후 이미지 뿐만 아니라 시계 타입등이 변경되면
Custom이지만 background는 Default일수도 있을 것 같아요.
그래서 체크를 안전하게 하면 좋을 것 같네요

```suggestion
            if (value is TmpScreenState.Custom) {
                saveWallPaperUseCase((value.screen.background as?LockScreenBackground.LocalImage).imageUri ?: return)
            }
```

Reply:
거기까진 생각 못했네요. 감사합니다 :)